controllers/replica_set_controller.go

```diff
@@ -611,8 +612,8 @@ func getCustomRolesModification(mdb mdbv1.MongoDBCommunity) (automationconfig.Mo
 	}, nil
 }
 
-func (r ReplicaSetReconciler) buildAutomationConfig(mdb mdbv1.MongoDBCommunity) (automationconfig.AutomationConfig, error) {
-	tlsModification, err := getTLSConfigModification(r.client, r.client, mdb)
+func (r ReplicaSetReconciler) buildAutomationConfig(ctx context.Context, mdb mdbv1.MongoDBCommunity) (automationconfig.AutomationConfig, error) {
```

Comment:
Which field exactly?

Reply:
I mean the context could be set as a member field in `ReplicaSetReconciler`. We don't need it passed in every private method as it will be the same context that we'll get in `Reconcile` method (or even passed in the "constructor")

Reply:
But it depends on the class usage. If the instance is shared then no, it's not possible to store ctx in a field. Here it seems it is one-shot instance in every reconcile process so it could work and would allow to have less changes.